packages/flutter/lib/src/widgets/platform_view.dart

```diff
@@ -358,47 +360,296 @@ class AppKitView extends _DarwinView {
+/// import 'package:web/web.dart' as web;
+///
+/// // Called after `element` is injected into the DOM.
+/// void onElementInjected(web.HTMLDivElement element) {
+///   final web.Element? located = web.document.querySelector('#someIdThatICanFindLater');
+///   assert(located == element, 'Wrong `element` located!');
+///   // Do things with `element` or `located`, or call your code now...
+///   element.style.backgroundColor = 'green';
+/// }
+///
+/// void onElementCreated(Object element) {
+///   element as web.HTMLDivElement;
+///   element.style.backgroundColor = 'red';
+///   element.id = 'someIdThatICanFindLater';
+///
+///   // Create the observer
+///   final web.ResizeObserver observer = web.ResizeObserver((
+///     JSArray<web.ResizeObserverEntry> entries,
+///     web.ResizeObserver observer,
+///   ) {
+///     if (element.isConnected) {
+///       // The observer is done, disconnect it.
+///       observer.disconnect();
+///       // Call our callback.
+///       onElementInjected(element);
+///     }
+///   }.toJS);
+///
+///   // Connect the observer.
+///   observer.observe(element);
+/// }
+/// ```
+///
+/// * Read more about [`ResizeObserver` in the MDN](https://developer.mozilla.org/en-US/docs/Web/API/Resize_Observer_API).
+///
+/// #### Other Observers
+///
+/// The example above uses a `ResizeObserver` because it can be applied directly
+/// to the `element` that is about to be injected. The most _correct_ observer
+/// for this (albeit requiring slightly more code) would be a
+/// [`MutationObserver`](https://developer.mozilla.org/en-US/docs/Web/API/MutationObserver).
+///
+/// The `MutationObserver` requires the "parent" element in which the
+/// `HtmlElementView` is going to be inserted. A safe way to retrieve a "parent"
+/// element for the platform views is to retrieve the `hostElement` of the
+/// [FlutterView] where the `HtmlElementView` is being rendered.
+///
+/// The `hostElement` of the current [FlutterView] can be retrieved through:
+///
+/// ```dart
+/// import 'dart:js_interop';
+/// import 'dart:ui_web' as ui_web;
+/// import 'package:flutter/widgets.dart';
+///
+/// void useHostElement(BuildContext context) {
+///   final int flutterViewId = View.of(context).viewId;
+///   final JSAny? hostElement = ui_web.views.getHostElement(flutterViewId);
+///   // Use `package:web` with `hostElement`...
+/// }
+/// ```
+///
+/// **Important:** If your flutter App uses `registerViewFactory` to register a
+/// factory function that creates your `HtmlElementView` widget, take into
+/// account that that the `viewId` parameter passed to your factory function
+/// **is not the same** as the `flutterViewId` obtained above:
+///
+/// * `flutterViewId` (from `View.of(context)`) represents the [FlutterView]
+///   where the web app is currently rendering.
+/// * `viewId` (passed to the `viewFactory` function) represents a unique ID
+///   for the `HtmlElementView` instance that is being injected into the app.
```

Comment:
I'd shorten this paragraph to just one sentence: "Note that `FlutterView.viewId` and the platform `viewId` passed to `RegisterViewFactory` identify different kinds of objects".

Reply:
Yep, short and sweet.